### PR TITLE
Add a prefer wayland option

### PR DESCRIPTION
**Pull request by YaLTeR**

### CODE REVIEW

File: MonoGame.Framework/Game.cs [hunk 1]
```diff
@@ -74,9 +74,12 @@ public bool WaylandVsync
             }
         }
 
+        // If set to true, try to initialize the Wayland SDL video platform first.
```
Comment (Swan) [inline MonoGame.Framework/Game.cs]:
```suggestion
        /// <summary>
        ///     If set to true, try to initialize the Wayland SDL video platform first.
        /// </summary>
```

File: MonoGame.Framework/Game.cs [hunk 2]
```diff
@@ -74,9 +74,12 @@ public bool WaylandVsync
...
+        internal bool PreferWayland { get; }
+
         partial void PlatformConstruct();
 
-        public Game()
+        public Game(bool preferWayland)
```
Comment (Swan) [inline MonoGame.Framework/Game.cs]: Should probably make this argument optional

```suggestion
        public Game(bool preferWayland = false)
```

**Review by Swan — CHANGES_REQUESTED** [review]

**Comment by lurr** [comment]
Would it be better to instead check the `XDG_SESSION_TYPE` environment variable here instead? This would remove the requirement for this argument and allow smarter selection of which SDL video subsystem to initialize.

**Comment by YaLTeR** [comment]
From what I recall, this isn't a standard environment variable. In any case, I'd rather simply make this the default behavior, but I'm waiting for SDL2 to implement CSD.

**Pull request merged**